### PR TITLE
[Site] update to Netlify hosting

### PR DESCRIPTION
**Pull request by riley**
### Description
This PR moves our hosting from GitHub Pages over to Netlify ✨.

connects #203 

The steps after this PR is merged are
1. Go into the netlify console and plug in our deploy scripts.
2. Make sure the builds work
3. Move DNS over

### Motivation and context
Some of the advantages over GH Pages:
- Deploy Previews (this is the big one. free staging site without a different environment config)
- rollbacks
- free ssl
- custom rewrites and redirects

### How to test

1. `npm run clean && npm install`
2. `npm run deploy:site`. The first time you run this, you'll have to authenticate your instance of `netlify-cli` on their site with the `mineral-ci` user.
3. once the deploy finishes, you'll get a link to a netlify site. The site should look the same
4. 404s redirect to `/`, which is done with a webpack-dev-server setting + a netlify config file depending on environment

### Types of changes
- refactor

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered
* [x] Documentation created or updated

### How does this PR make you feel?
Alt text here

### CODE REVIEW

File: packages/site/src/components/NoMatch.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    fontSize: '10em',
+    fontWeight: 'bold',
+    margin: '1em 0 20px'
+  }
+};
+
+const Error = createStyledComponent('p', styles.error);
+const Container = createStyledComponent('div', styles.container);
+
+export default function NoMatch() {
```
Comment (brentertz) [inline packages/site/src/components/NoMatch.js]: Perhaps `NotFound` might be more clean than `NoMatch`.   More importantly, why is this needed and how does it work, if 404's are supposed to redirect to the home page?

File: package.json [hunk 2]
```diff
@@ -77,6 +77,7 @@
     "lodash.upperfirst": "4.3.1",
     "material-design-icons": "3.0.1",
     "mustache": "2.3.0",
+    "netlify-cli": "1.2.2",
```
Comment (brentertz) [inline package.json]: The `gh-pages` package is no longer needed and can be removed.

File: packages/site/src/web/index.js [hunk 3]
```diff
@@ -31,7 +31,8 @@ require('react-live/react-live.css');
 // Enable Glamor simulate helper
 simulations(true);
 
-// Github 404 page hack to support SPA hosting
```
Comment (brentertz) [inline packages/site/src/web/index.js]: Related, I can't imagine that `site/public/404.html` still needed.  That file is a github-pages specific thing.

Comment (brentertz) [inline packages/site/src/web/index.js]: Continuing on, is the `site/public/CNAME` file still needed?  Does netlify use that too?

File: .netlify [hunk 4]
```diff
@@ -0,0 +1 @@
+{"site_id":"d7bf54c3-d97a-44fe-a810-fdfdb42ace3d","path":"packages/site/dist"}
```
Comment (riley) [inline .netlify]: I don't think it _needs_ to be secure. There is another generated file that lives in `$HOME/netlify` that contains an access token specific to your machine. This site id is just the pointer to our `http://<rando-hash>.stenographer-panda-45372.netlify.com` domain. Since only our team is going to deploy the site however, I think it'd be fine if this file is kept out of the repo. I'll trust your judgement on this one.

File: package.json (hunk 2 above)
Comment (riley) [inline package.json]: ah, of course

**Comment by coldpour** [comment]
looks like "How to Test" got munged with the "Checklist"

File: package.json [hunk 5]
```diff
@@ -9,8 +9,8 @@
     "clean": "find packages -name dist -or -name lib -or -name node_modules -type d | xargs rm -rf",
     "commit": "git-cz",
     "commitmsg": "validate-commit-msg",
-    "deploy:site": "npm run build && npm run test && npm run deploy:site:dist",
-    "deploy:site:dist": "gh-pages -d packages/site/dist -m 'chore(site): Deploy' -r https://github.com/mineral-ui/mineral-ui.github.io.git -b master",
+    "deploy:site": "npm run build && npm run test && npm run netlify",
+    "netlify": "netlify deploy",
```
Comment (coldpour) [inline package.json]: we may need two scripts for a while to deploy both sites in parallel until we get DNS migrated successfully

File: .netlify (hunk 4 above)
Comment (coldpour) [inline .netlify]: it looks like it's just a uuid, which doesn't need to be secure

File: packages/site/src/components/NoMatch.js (hunk 1 above)
Comment (riley) [inline packages/site/src/components/NoMatch.js]: ah, I thought we still wanted to have 404s for some reason. I can remove all this.

**Comment by riley** [comment]
@coldpour I think keeping the `gh-pages` stuff in `master` is related to [this comment](https://github.com/mineral-ui/mineral-ui/pull/243#issuecomment-321655653) on the other site pr

File: packages/site/src/web/index.js (hunk 3 above)
Comment (riley) [inline packages/site/src/web/index.js]: nope, removed

File: package.json (hunk 2 above)
Comment (riley) [inline package.json]: update: keeping this for now until the site gets to MVP; need to retain ability to push components out to the world.

File: packages/site/src/components/Nav.js [hunk 6]
```diff
@@ -45,6 +45,9 @@ const styles = {
     margin: '0',
     fontSize: theme.fontSize_h4
   }),
+  link: {
+    textDecoration: 'none'
```
Comment (brentertz) [inline packages/site/src/components/Nav.js]: Is this needed anymore, following the latest Link update?  https://github.com/mineral-ui/mineral-ui/blob/master/packages/link/src/Link.js#L57

**Comment by brentertz** [comment]
Besides the obvious rebasing that needs done, are you sure that you have the order of events correct?  I would assume that you need to have the site deployed to netlify (and have the team test it), then switch the DNS over, then merge this PR.  I only suggest this as DNS can sometimes take a while to transfer and merging this PR will temporarily break our ability to make public updates to the site.

**Comment by riley** [comment]
cherrypicking my commit and closing in favor of #268 because this got stale
